internal/services/channel_account_service.go

```diff
@@ -0,0 +1,150 @@
+		return fmt.Errorf("inserting channel accounts: %w", err)
+	}
+
+	return nil
+}
+
+func (s *channelAccountService) submitCreateChannelAccountsOnChainTransaction(ctx context.Context, distributionAccountPublicKey string, ops []txnbuild.Operation) error {
+	distributionAccount, err := s.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: distributionAccountPublicKey})
+	if err != nil {
+		return fmt.Errorf("getting account detail of channel account: %w", err)
+	}
+
+	tx, err := txnbuild.NewTransaction(
+		txnbuild.TransactionParams{
+			SourceAccount:        &distributionAccount,
+			IncrementSequenceNum: true,
+			Operations:           ops,
+			BaseFee:              s.BaseFee,
+			Preconditions:        txnbuild.Preconditions{TimeBounds: txnbuild.NewTimeout(300)},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("building transaction: %w", err)
+	}
+
+	signedTx, err := s.DistributionAccountSignatureClient.SignStellarTransaction(ctx, tx, distributionAccountPublicKey)
+	if err != nil {
+		return fmt.Errorf("signing transaction: %w", err)
+	}
+
+	hash, err := signedTx.Hash(s.DistributionAccountSignatureClient.NetworkPassphrase())
+	if err != nil {
+		return fmt.Errorf("getting transaction hash: %w", err)
+	}
+
+	_, err = s.HorizonClient.SubmitTransaction(signedTx)
+	if err != nil {
+		if hError := horizonclient.GetError(err); hError != nil {
+			hProblem := hError.Problem
+			if hProblem.Type == "https://stellar.org/horizon-errors/timeout" {
+				return fmt.Errorf("horizon request timed out while creating a channel account. Transaction hash: %s", hash)
+			}
+
+			errString := fmt.Sprintf("Type: %s, Title: %s, Status: %d, Detail: %s, Extras: %v", hProblem.Type, hProblem.Title, hProblem.Status, hProblem.Detail, hProblem.Extras)
+			return fmt.Errorf("submitting transaction: %s: %w", errString, err)
+		} else {
+			return fmt.Errorf("submitting transaction: %w", err)
+		}
+	}
+
+	return nil
+}
+
+type ChannelAccountServiceOptions struct {
+	DB                                 db.ConnectionPool
+	HorizonClient                      horizonclient.ClientInterface
+	BaseFee                            int64
+	DistributionAccountSignatureClient signing.SignatureClient
+	ChannelAccountStore                channelaccounts.ChannelAccountStore
+	PrivateKeyEncrypter                channelaccounts.PrivateKeyEncrypter
+	EncryptionPassphrase               string
+}
+
+func NewChannelAccountService(opts ChannelAccountServiceOptions) *channelAccountService {
+	return &channelAccountService{
+		DB:                                 opts.DB,
+		HorizonClient:                      opts.HorizonClient,
+		BaseFee:                            opts.BaseFee,
+		DistributionAccountSignatureClient: opts.DistributionAccountSignatureClient,
+		ChannelAccountStore:                opts.ChannelAccountStore,
+		PrivateKeyEncrypter:                opts.PrivateKeyEncrypter,
+		EncryptionPassphrase:               opts.EncryptionPassphrase,
+	}
+}
```

Comment:
polishing: what's the advantage of such constructor if all you're doing is assign the variables one by one?